add JSON interface for generateOsgUserVoMap (SOFTWARE-2380)

edquist opened the pull request.

edquist committed:
add JSON interface for generateOsgUserVoMap (SOFTWARE-2380)

bbockelm commented on gums-service/src/main/java/org/opensciencegrid/gums/service/JSONMapper.java:
```diff
@@ -105,6 +109,49 @@ private void doVersion(HttpServletRequest request,
+        Configuration configuration = null;
+        try
+        {
+            configuration = gums.getConfiguration();
```
Why is the `Configuration` object needed?  Doesn't seem to be used...

edquist replied:
This was a mistake - I was following the pattern of the other doX functions here and didn't notice it was unused.  I'll remove it.

bbockelm commented on gums-service/src/main/java/org/opensciencegrid/gums/service/JSONMapper.java:
```diff
@@ -105,6 +109,49 @@ private void doVersion(HttpServletRequest request,
+        String map = null;
+        try
+        {
+            map = gums.generateOsgUserVoMap(hostname);
```
The `generateOsgUserVoMap` method should be applying access controls (basically, the DN of the hostcert should equal the hostname).  Have you verified this is true?

edquist replied:
Yeah, I saw that `generateOsgUserVoMap` tests for `hasReadAllAccess(currentUser(), hostname)`, here:
https://github.com/opensciencegrid/gums/blob/v1.5.2/gums-core/src/main/java/gov/bnl/gums/admin/GUMSAPIImpl.java#L236

But if the access requirements should be done differently let me know.

bbockelm replied:
Nope - the current method is fine.  I just wanted to make sure you had tested this aspect to make sure it is working.

edquist replied:
Ok, no I have not done manual tests of this.

bbockelm commented:
Ok - I'll probably hold off on merging until there's a prototype client we can use to exercise the interface a bit.  However, everything looks from from my POV.

edquist committed:
new function to return json map instead of file contents (SOFTWARE-2380)

... fairly uncreatively named "getOsgUserVoMap" instead of the existing
"generateOsgUserVoMap", which returned a string representing the
contents of the osg user vo map file.

We may in fact want to get rid of the json api for generateOsgUserVoMap
and possibly rename the new getOsgUserVoMap to some better name.

edquist committed:
change name to reflect reversed direction of map (SOFTWARE-2380)

edquist committed:
omit VOs with no valid user names (SOFTWARE-2380)

bbockelm commented:
@edquist - I'm cleaning up GUMS requests.  Is this still ready to go?

edquist commented:
Yeah, we currently ship these changes as sw-2380-json-user-vo-map.patch
